### PR TITLE
Filter tarball files with globs; filter more files

### PR DESCRIPTION
**Pull request by V-FEXrt**
Part two of splitting up https://github.com/sifive/wake/pull/1364

This change 
 - removes more files from the  tarball that have snuck in over time
 - uses globs  for exclusion instead of regexes
 - prints the excluded files to `info`

### CODE REVIEW

File: share/wake/lib/core/regexp.wake [hunk 1]
```diff
@@ -81,6 +81,28 @@ export def regExpToString (regExp: RegExp): String =
...
+# Test if a String matches all provided regular expressions
+# matchesAll (`a*`, `*b`) "ab" = True
+# matchesAll (`a*`, `*b`) "a" = False
+export def matchesAll (testRegExps: List RegExp) (str: String): Boolean = match testRegExps
+    Nil = True
+    head, tail =
+        if (matches head str) then
+            matchesAll tail str
+        else
+            False
```
Comment (JakeSiFive) [inline share/wake/lib/core/regexp.wake:93]: This is too generic to be in regex, use forall/exist in list.wake

File: share/wake/lib/core/regexp.wake [hunk 2]
```diff
@@ -81,6 +81,28 @@ export def regExpToString (regExp: RegExp): String =
...
+
+# Test if a String matches any provided regular expressions
+# matchesAny (`a*`, `*b`) "ac" = True
+# matchesAny (`a*`, `*b`) "c" = False
+export def matchesAny (testRegExps: List RegExp) (str: String) = match testRegExps
```
Comment (JakeSiFive) [inline share/wake/lib/core/regexp.wake:98]: ditto

File: share/wake/lib/core/regexp.wake (hunk 1 above)
Comment (V-FEXrt) [reply share/wake/lib/core/regexp.wake]: ah ye. makes sense. fixed!

**Review by ag-eitilt — APPROVED** [review]

**Comment by V-FEXrt** [comment]
@JakeSiFive I'm going to go ahead and land this since I've addressed your comments and this PR is fairly small/uncontroversial now. If you have any changes you want when you get  back lmk and I'll make them asap

**Pull request merged**